### ISSUE TITLE
Eventual DynDNS fail

### ISSUE THREAD

**Issue by mray** [freedombox/Plinth#496]
The "Status" tab on the DynDNS tab indicates no successful updates:

> **NAT type**
Behind NAT. This means that Dynamic DNS service will poll the "IP Check URL" for changes (the "IP Check URL" entry is needed for this, otherwise IP changes will not be detected). In case the WAN IP changes, it may take up to 5 minutes until your DNS entry is updated.
**Last update**
no successful update recorded since last config change 

But `host freedombox.domain.de` and `curl ipinfo.io` point to the same IP.
Running into lots of other issues I wonder if something fishy is going on.
The app should be clear about whether it is set up correctly or not with a status icon similar to "failed "passed" and "error" tags on diagnostics elsewhere.

How can I be sure if everything *is* set up right?

**Comment by SunilMohanAdapa** [comment]
Is your FreedomBox behind a router?  If so, please forward port 80 and 443 from the router to FreedomBox.

**Comment by mray** [comment]
It is behind a router (of course ;) )and I forward port 443. I do *not* forward port port 80 since I assumed that I want exclusively use https all the time anyway.

If I need to forward port 80, how would lets encrypt know what port to use from the outside? How does it even know where current port 443 is forwarded from? I never provided that info - maybe that is the problem?

**Comment by SunilMohanAdapa** [comment]
On 06/10/2016 01:17 PM, mray wrote:
> It is behind a router (of course ;) )and I forward port 443. I do /not/
> forward port port 80 since I assumed that I want exclusively use https
> all the time anyway.

Let's encrypt requires port 80 to obtain a certificate (and to keep
renewing it).

> 
> If I need to forward port 80, how would lets encrypt know what port to
> use from the outside? How does it even know where current port 443 is
> forwarded from? I never provided that info - maybe that is the problem?

Lets encrypt will try a URL like
http://freedombox.domain.de/.well-known/...   This means port 80.  Which
will land on port 80 of the router.  That if redirected to port 80 of
FreedomBox, will succeed in obtaining the certificate.

I suppose we also need to explain in the UI to our users who are behind NAT.

-- 
Sunil

**Comment by mray** [comment]
even after generating a Let's Encrypt  cert and forwarding port 80 and 443 directly to the freedombox (and a reboot) DynDNS still indicates no successful updates.

I did wait longer than 5 minutes and I did provide the IP check url http://myip.datasystems24.de

**Comment by steglicd** [comment]
what happens if you enter your update URL into your Browser?
What happens if you run: `bash -x /usr/share/plinth/actions/dynamicdns update` ?
Do you have a file called `/etc/cron.d/ez-ipupdate` ?

**Comment by mray** [comment]
> what happens if you enter your update URL into your Browser?

I can read my current IP. (same as FreedomBox seems to have)

> What happens if you run: bash -x /usr/share/plinth/actions/dynamicdns update ?

A whole lot of stuff scrolls by but the status message of the DynDNS remains.

> Do you have a file called /etc/cron.d/ez-ipupdate ?

`cat /etc/cron.d/ez-ipupdate` returned `*/5 * * * * root /usr/share/plinth/actions/dynamicdns update` that means it exists I think.

**Comment by steglicd** [comment]
Did you enter the " update URL" into your browser? Or the "IP check url"? Can you send me the output of the `bash -x /usr/share/plinth/actions/dynamicdns update` command (remove passwords please)?

**Comment by mray** [comment]
> Did you enter the " update URL" into your browser? Or the "IP check url"?

Sorry I checked the wrong one. The update url asks for credentials, then shows "notfqdn" and my IP. 

```
~$` bash -x /usr/share/plinth/actions/dynamicdns update
+ PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
++ which wget
+ WGET=/usr/bin/wget
+ WGETOPTIONS='-4 -o /dev/null -t 3 -T 3'
+ EMPTYSTRING=none
+ NOIP=0.0.0.0
+ UPDATEMINUTES=5
+ UPDATEMINUTESUNKNOWN=3600
+ TOOLNAME=ez-ipupdate
++ which ez-ipupdate
+ UPDATE_TOOL=/usr/sbin/ez-ipupdate
+ DISABLED_STRING=disabled
+ ENABLED_STRING=enabled
+ CFGDIR=/etc/ez-ipupdate/
+ CFG=/etc/ez-ipupdate/ez-ipupdate.conf
+ CFG_disabled=/etc/ez-ipupdate/ez-ipupdate.inactive
+ IPFILE=/etc/ez-ipupdate/ez-ipupdate.currentIP
+ STATUSFILE=/etc/ez-ipupdate/ez-ipupdate.status
+ LASTUPDATE=/etc/ez-ipupdate//last-update
+ HELPERCFG=/etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ CRONJOB=/etc/cron.d/ez-ipupdate
+ PIDFILE=/var/run/ez-ipupdate.pid
+ cmd=update
+ shift
+ cfgfile=/tmp/none
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.inactive ']'
+ cfgfile=/etc/ez-ipupdate/ez-ipupdate.inactive
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.conf ']'
+ case ${cmd} in
+ doReadCFG
+ host=
+ server=
+ user=
+ pass=
+ ipurl=
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate.inactive ']'
++ grep '^host=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
+ host=freedombox.****.de
++ cut -d = -f 2-
++ grep -v '^'\'''\'''
++ grep '^server=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ server=
++ cut -d = -f 2-
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d : -f 1
+ user=****
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d : -f 2-
++ cut -d = -f 2-
+ pass=****
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate-plinth.cfg ']'
++ awk '{print $2}'
++ grep -v '^'\'''\'''
++ grep '^IPURL' /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ ipurl=http://myip.datasystems24.de
++ grep POSTURL /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ awk '{print $2}'
++ grep -v '^'\'''\'''
+ updateurl='http://dyndns.strato.com/nic/update/?hostname=<Domain>'
++ awk '{print $2}'
++ grep POSTAUTH /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
+ basicauth=enabled
++ grep POSTSSLIGNORE /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ ignoreCertError=disabled
++ nslookup freedombox.****.de
++ tail -n2
++ grep A
++ sed 's/[^0-9.]//g'
+ dnsentry=46.232.103.199
+ doGetWANIP
+ '[' '!' -z http://myip.datasystems24.de ']'
++ mktemp
+ outfile=/tmp/user/10000/tmp.F4hhJhmkw0
+ local 'cmd=/usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/10000/tmp.F4hhJhmkw0 http://myip.datasystems24.de'
+ /usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/10000/tmp.F4hhJhmkw0 http://myip.datasystems24.de
++ sed 's/[^0-9.]//g' /tmp/user/10000/tmp.F4hhJhmkw0
+ wanip=46.232.103.199
+ rm /tmp/user/10000/tmp.F4hhJhmkw0
+ '[' -z 46.232.103.199 ']'
+ echo 46.232.103.199
/usr/share/plinth/actions/dynamicdns: line 367: /etc/ez-ipupdate/ez-ipupdate.currentIP: Permission denied
+ grep -v execute /etc/ez-ipupdate/ez-ipupdate.inactive
/usr/share/plinth/actions/dynamicdns: line 368: /etc/ez-ipupdate/ez-ipupdate.inactive.tmp: Permission denied
+ mv /etc/ez-ipupdate/ez-ipupdate.inactive.tmp /etc/ez-ipupdate/ez-ipupdate.inactive
mv: cannot stat '/etc/ez-ipupdate/ez-ipupdate.inactive.tmp': No such file or directory
+ echo 'execute=/usr/share/plinth/actions/dynamicdns success 46.232.103.199'
/usr/share/plinth/actions/dynamicdns: line 370: /etc/ez-ipupdate/ez-ipupdate.inactive: Permission denied
+ '[' 46.232.103.199 '!=' 46.232.103.199 -a 46.232.103.199 '!=' 0.0.0.0 ']'
+ '[' 46.232.103.199 = 0.0.0.0 ']'
+ exit 0

```

**Comment by steglicd** [comment]
Please run this as root user.

**Comment by mray** [comment]
```
$ sudo bash -x /usr/share/plinth/actions/dynamicdns update
[sudo] password for mray: 
+ PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
++ which wget
+ WGET=/usr/bin/wget
+ WGETOPTIONS='-4 -o /dev/null -t 3 -T 3'
+ EMPTYSTRING=none
+ NOIP=0.0.0.0
+ UPDATEMINUTES=5
+ UPDATEMINUTESUNKNOWN=3600
+ TOOLNAME=ez-ipupdate
++ which ez-ipupdate
+ UPDATE_TOOL=/usr/sbin/ez-ipupdate
+ DISABLED_STRING=disabled
+ ENABLED_STRING=enabled
+ CFGDIR=/etc/ez-ipupdate/
+ CFG=/etc/ez-ipupdate/ez-ipupdate.conf
+ CFG_disabled=/etc/ez-ipupdate/ez-ipupdate.inactive
+ IPFILE=/etc/ez-ipupdate/ez-ipupdate.currentIP
+ STATUSFILE=/etc/ez-ipupdate/ez-ipupdate.status
+ LASTUPDATE=/etc/ez-ipupdate//last-update
+ HELPERCFG=/etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ CRONJOB=/etc/cron.d/ez-ipupdate
+ PIDFILE=/var/run/ez-ipupdate.pid
+ cmd=update
+ shift
+ cfgfile=/tmp/none
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.inactive ']'
+ cfgfile=/etc/ez-ipupdate/ez-ipupdate.inactive
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.conf ']'
+ case ${cmd} in
+ doReadCFG
+ host=
+ server=
+ user=
+ pass=
+ ipurl=
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate.inactive ']'
++ cut -d = -f 2-
++ grep '^host=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ host=freedombox.domain.de
++ grep -v '^'\'''\'''
++ cut -d = -f 2-
++ grep '^server=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ server=
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d : -f 1
++ cut -d = -f 2-
+ user=mray.de
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
++ cut -d : -f 2-
+ pass=****
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate-plinth.cfg ']'
++ grep '^IPURL' /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ awk '{print $2}'
++ grep -v '^'\'''\'''
+ ipurl=http://myip.datasystems24.de
++ grep POSTURL /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ updateurl='http://dyndns.strato.com/nic/update/?hostname=<Domain>'
++ awk '{print $2}'
++ grep POSTAUTH /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
+ basicauth=enabled
++ grep POSTSSLIGNORE /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ ignoreCertError=disabled
++ tail -n2
++ grep A
++ nslookup freedombox.domain.de
++ sed 's/[^0-9.]//g'
+ dnsentry=46.232.103.199
+ doGetWANIP
+ '[' '!' -z http://myip.datasystems24.de ']'
++ mktemp
+ outfile=/tmp/user/0/tmp.30sVx56pHU
+ local 'cmd=/usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.30sVx56pHU http://myip.datasystems24.de'
+ /usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.30sVx56pHU http://myip.datasystems24.de
++ sed 's/[^0-9.]//g' /tmp/user/0/tmp.30sVx56pHU
+ wanip=46.232.103.199
+ rm /tmp/user/0/tmp.30sVx56pHU
+ '[' -z 46.232.103.199 ']'
+ echo 46.232.103.199
+ grep -v execute /etc/ez-ipupdate/ez-ipupdate.inactive
+ mv /etc/ez-ipupdate/ez-ipupdate.inactive.tmp /etc/ez-ipupdate/ez-ipupdate.inactive
+ echo 'execute=/usr/share/plinth/actions/dynamicdns success 46.232.103.199'
+ '[' 46.232.103.199 '!=' 46.232.103.199 -a 46.232.103.199 '!=' 0.0.0.0 ']'
+ '[' 46.232.103.199 = 0.0.0.0 ']'
+ exit 0

```

**Comment by mray** [comment]
```
$ sudo bash -x /usr/share/plinth/actions/dynamicdns update
[sudo] password for mray: 
+ PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
++ which wget
+ WGET=/usr/bin/wget
+ WGETOPTIONS='-4 -o /dev/null -t 3 -T 3'
+ EMPTYSTRING=none
+ NOIP=0.0.0.0
+ UPDATEMINUTES=5
+ UPDATEMINUTESUNKNOWN=3600
+ TOOLNAME=ez-ipupdate
++ which ez-ipupdate
+ UPDATE_TOOL=/usr/sbin/ez-ipupdate
+ DISABLED_STRING=disabled
+ ENABLED_STRING=enabled
+ CFGDIR=/etc/ez-ipupdate/
+ CFG=/etc/ez-ipupdate/ez-ipupdate.conf
+ CFG_disabled=/etc/ez-ipupdate/ez-ipupdate.inactive
+ IPFILE=/etc/ez-ipupdate/ez-ipupdate.currentIP
+ STATUSFILE=/etc/ez-ipupdate/ez-ipupdate.status
+ LASTUPDATE=/etc/ez-ipupdate//last-update
+ HELPERCFG=/etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ CRONJOB=/etc/cron.d/ez-ipupdate
+ PIDFILE=/var/run/ez-ipupdate.pid
+ cmd=update
+ shift
+ cfgfile=/tmp/none
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.inactive ']'
+ cfgfile=/etc/ez-ipupdate/ez-ipupdate.inactive
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.conf ']'
+ case ${cmd} in
+ doReadCFG
+ host=
+ server=
+ user=
+ pass=
+ ipurl=
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate.inactive ']'
++ cut -d = -f 2-
++ grep '^host=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ host=freedombox.domain.de
++ grep -v '^'\'''\'''
++ cut -d = -f 2-
++ grep '^server=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ server=
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d : -f 1
++ cut -d = -f 2-
+ user=mray.de
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
++ cut -d : -f 2-
+ pass=****
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate-plinth.cfg ']'
++ grep '^IPURL' /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ awk '{print $2}'
++ grep -v '^'\'''\'''
+ ipurl=http://myip.datasystems24.de
++ grep POSTURL /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ updateurl='http://dyndns.strato.com/nic/update/?hostname=<Domain>'
++ awk '{print $2}'
++ grep POSTAUTH /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
+ basicauth=enabled
++ grep POSTSSLIGNORE /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ ignoreCertError=disabled
++ tail -n2
++ grep A
++ nslookup freedombox.domain.de
++ sed 's/[^0-9.]//g'
+ dnsentry=46.232.103.199
+ doGetWANIP
+ '[' '!' -z http://myip.datasystems24.de ']'
++ mktemp
+ outfile=/tmp/user/0/tmp.30sVx56pHU
+ local 'cmd=/usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.30sVx56pHU http://myip.datasystems24.de'
+ /usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.30sVx56pHU http://myip.datasystems24.de
++ sed 's/[^0-9.]//g' /tmp/user/0/tmp.30sVx56pHU
+ wanip=46.232.103.199
+ rm /tmp/user/0/tmp.30sVx56pHU
+ '[' -z 46.232.103.199 ']'
+ echo 46.232.103.199
+ grep -v execute /etc/ez-ipupdate/ez-ipupdate.inactive
+ mv /etc/ez-ipupdate/ez-ipupdate.inactive.tmp /etc/ez-ipupdate/ez-ipupdate.inactive
+ echo 'execute=/usr/share/plinth/actions/dynamicdns success 46.232.103.199'
+ '[' 46.232.103.199 '!=' 46.232.103.199 -a 46.232.103.199 '!=' 0.0.0.0 ']'
+ '[' 46.232.103.199 = 0.0.0.0 ']'
+ exit 0

```

**Comment by steglicd** [comment]
Looks good. But still no update success message is displayed?

**Comment by mray** [comment]
no. 
![image](https://cloud.githubusercontent.com/assets/164785/15979227/218eb214-2f63-11e6-903a-de59d401df4d.png)

**Comment by steglicd** [comment]
Ah... ok... no Update is needed in your case. For this reason no update is done. Change your WAN IP and try again  please.

**Comment by mray** [comment]
>Change your WAN IP and try again please.

I think this can be done by my ISP, no idea how I can do that.

If I don't need an update does that mean everything is working fine and DynDNS **isn't** a reason why radicale won't work correctly? I guess my issue is that the status information isn't informative to me as a user. I have no clue about what equates good, bad or good enough information.
Having a three level indicator: ON!, off, error might be a way to clear things at least to some degree.

**Comment by mray** [comment]
I found a way to change my IP from the interface from my router. Now the Status page says
> last update done (Sat Jun 11 09:25:03 UTC 2016)

So what was the problem?

**Comment by steglicd** [comment]
I think there was no problem at all. The DynDNS module monitors your current IP address and compares the result with the currently registered DNS record. If the record does not match the current IP the Module will update the DNS record. 

In your case the record was already up to date. So no need to do an update. And because there was no update done, there was not successful last update recorded. 

Maybe we could improve the status message here.

**Comment by mray** [comment]
I managed to break things again but don't know how. I suspect it is the mapping of port 80 and 443 locally to 80 and 443 globally. Here is what I did:

* changed port mapping
* discovered that https isn't working anymore
* revoked my Let's Encrypt cert and *tried* to re-obtain but failed
* renewed my IP
* ran `bash -x /usr/share/plinth/actions/dynamicdns update`
* compared IPs of `curl ipinfo.io` vs. `host freedombox.mray.de` -> different IPs.

What do I need to do now?

**Comment by steglicd** [comment]
After IP renewal it should not be needed to run `bash -x /usr/share/plinth/actions/dynamicdns update`, just wait for up to 5 minutes. 

What did `bash -x /usr/share/plinth/actions/dynamicdns update` printed out?

**Comment by steglicd** [comment]
I added a pull request to improve status message in case of no update is needed. So the "no update recorded" message should disappear in case of everything is fine and no update was done due to this fact.

**Comment by mray** [comment]
I noticed that "Use HTTP basic authentication" wasn't marked, when it is supposed to be.
I fixed that and renewed my IP. After that I rebooted.
Now `host freedombox.domain.de` and curl `ipinfo.io` show the same ip again and the DynDNS Status says good.
But whenever I try to re-obtain my Let's Encrypt certificate it complains:


```
 Failed to obtain certificate for domain freedombox.domain.de: Failed authorization procedure. freedombox.domain.de (http-01): urn:acme:error:connection :: The server could not connect to the client to verify the domain :: Could not connect to http://freedombox.domain.de/.well-known/acme-challenge/_Gw8czMR4Z9bNGLjlfwA5U8sn6ibIP0W13Noi7SMpIw 
```

I also can't seem to reach https://freedombox.domain.de/radicale.

here is what `sudo bash -x /usr/share/plinth/actions/dynamicdns update` says:

```
$ sudo bash -x /usr/share/plinth/actions/dynamicdns update
[sudo] password for mray: 
+ PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
++ which wget
+ WGET=/usr/bin/wget
+ WGETOPTIONS='-4 -o /dev/null -t 3 -T 3'
+ EMPTYSTRING=none
+ NOIP=0.0.0.0
+ UPDATEMINUTES=5
+ UPDATEMINUTESUNKNOWN=3600
+ TOOLNAME=ez-ipupdate
++ which ez-ipupdate
+ UPDATE_TOOL=/usr/sbin/ez-ipupdate
+ DISABLED_STRING=disabled
+ ENABLED_STRING=enabled
+ CFGDIR=/etc/ez-ipupdate/
+ CFG=/etc/ez-ipupdate/ez-ipupdate.conf
+ CFG_disabled=/etc/ez-ipupdate/ez-ipupdate.inactive
+ IPFILE=/etc/ez-ipupdate/ez-ipupdate.currentIP
+ STATUSFILE=/etc/ez-ipupdate/ez-ipupdate.status
+ LASTUPDATE=/etc/ez-ipupdate//last-update
+ HELPERCFG=/etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ CRONJOB=/etc/cron.d/ez-ipupdate
+ PIDFILE=/var/run/ez-ipupdate.pid
+ cmd=update
+ shift
+ cfgfile=/tmp/none
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.inactive ']'
+ cfgfile=/etc/ez-ipupdate/ez-ipupdate.inactive
+ '[' -f /etc/ez-ipupdate/ez-ipupdate.conf ']'
+ case ${cmd} in
+ doReadCFG
+ host=
+ server=
+ user=
+ pass=
+ ipurl=
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate.inactive ']'
++ grep '^host=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
+ host=freedombox.domain.de
++ grep '^server=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
++ grep -v '^'\'''\'''
+ server=
++ cut -d : -f 1
++ cut -d = -f 2-
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
+ user=****
++ grep '^user=' /etc/ez-ipupdate/ez-ipupdate.inactive
++ cut -d = -f 2-
++ cut -d : -f 2-
+ pass=****
+ '[' '!' -z /etc/ez-ipupdate/ez-ipupdate-plinth.cfg ']'
++ grep '^IPURL' /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ grep -v '^'\'''\'''
++ awk '{print $2}'
+ ipurl=http://myip.datasystems24.de
++ awk '{print $2}'
++ grep -v '^'\'''\'''
++ grep POSTURL /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
+ updateurl='http://dyndns.strato.com/nic/update/?hostname=<Domain>'
++ grep POSTAUTH /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ awk '{print $2}'
++ grep -v '^'\'''\'''
+ basicauth=enabled
++ grep POSTSSLIGNORE /etc/ez-ipupdate/ez-ipupdate-plinth.cfg
++ awk '{print $2}'
++ grep -v '^'\'''\'''
+ ignoreCertError=disabled
++ nslookup freedombox.domain.de
++ tail -n2
++ grep A
++ sed 's/[^0-9.]//g'
+ dnsentry=149.172.54.196
+ doGetWANIP
+ '[' '!' -z http://myip.datasystems24.de ']'
++ mktemp
+ outfile=/tmp/user/0/tmp.48HpTd4Ahs
+ local 'cmd=/usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.48HpTd4Ahs http://myip.datasystems24.de'
+ /usr/bin/wget -4 -o /dev/null -t 3 -T 3 -O /tmp/user/0/tmp.48HpTd4Ahs http://myip.datasystems24.de
++ sed 's/[^0-9.]//g' /tmp/user/0/tmp.48HpTd4Ahs
+ wanip=149.172.54.196
+ rm /tmp/user/0/tmp.48HpTd4Ahs
+ '[' -z 149.172.54.196 ']'
+ echo 149.172.54.196
+ grep -v execute /etc/ez-ipupdate/ez-ipupdate.inactive
+ mv /etc/ez-ipupdate/ez-ipupdate.inactive.tmp /etc/ez-ipupdate/ez-ipupdate.inactive
+ echo 'execute=/usr/share/plinth/actions/dynamicdns success 149.172.54.196'
+ '[' 149.172.54.196 '!=' 149.172.54.196 -a 149.172.54.196 '!=' 0.0.0.0 ']'
+ '[' 149.172.54.196 = 0.0.0.0 ']'
+ exit 0

```

**Comment by steglicd** [comment]
I don't think this is dynDNS related. How do you try to reach `https://freedombox.domain.de/radicale`? From a PC within your LAN? Try from another Internet access (like mobile phone). Because some routers does not support NAT Hair pinning (access portforward from LAN side). Maybe you are to fast and the DNS entry is not timed out yet? So Let's encrypt still have a old DNS entry?

**Comment by mray** [comment]
> How do you try to reach https://freedombox.domain.de/radicale?

I tried from the same LAN but also over a VPN connection.

I don't know what access port forward from LAN side means exactly. But maybe having two routers does not work like I expect it:
I have a "FritzBox" from my ISP that is set up with an 'exposed host'  that is an "OpenWRT" router.
My guess is that it should work since port forwarding seems to have worked in the past. I was able to get a Let's Encrypt cert after all.

> Maybe you are to fast and the DNS entry is not timed out yet?

I waited quite an amount of time by now


> So Let's encrypt still have a old DNS entry?

Its diagnostic tool currently says:
`Access URL https://freedombox.domain.de 	*failed*`

**Comment by steglicd** [comment]
So you have a freedombox behind a OpenWRT box behind a Fritzbox? Wow. Maybe a more simple setup would be nice for debugging. 

So if you can reach http://<LAN IP of your freedom box> (from your LAN) but not http://freedombox.domain.de from the outside world it must be related to one of your routers (portforwarding).

And if this is the case, Let's encrypt will not work.

**Comment by mray** [comment]
Well I don't trust the FritzBox to build my local lan, but need it since it contains the cable modem. so set up an openwrt instead that gets "everything" routed to it without firewall. After that it should be just like behind any other router.

Do you use FreedomBox as router?

**Comment by steglicd** [comment]
I also don't trust Fritzboxes ;)

Can't you just operate the Fritzbox a s a simple modem? I just want to kick at least one possible cause for the error out of the NAT chain (like: Fredombox behind NAT behind NAT).

Because one of this boxes must be blocking the incoming requests (as I assume you can reach Freedombox from LAN but not from WAN).

**Comment by SunilMohanAdapa** [comment]
Got automatically closed due to merged pull request from @steglicd .  Reopening due to ongoing conversation.  Please close later.

**Comment by mray** [comment]
> Can't you just operate the FritzBox a s a simple modem?

"Exposed host" is what comes closest as far as I know. If it is possible to use a FritzBox *only* as modem that would be great.

>   (as I assume you can reach Freedombox from LAN but not from WAN).

I can reach it via the local 192.168.x.x IP, but the domain name does not resolve locally either.
What boggles me is that when I forwarded to port 443 from a strange port like 28963 least https://domain was somewhat reachable. But back when that worked i didn't even care about port 80 (which is why Let's Encrypt didn't work).

I'm profoundly confused and fear that I am at least *somwhat* representative of what we face as target audience in terms of needs an expectations...

**Comment by steglicd** [comment]
Ok, I think we need to do a step back, take a deep breath and investigate from the beginning.

1. could you please connect the Freedombox directly to the Fritzbox
2. freedombox should obtain a IP via DHCP
3. set Fritzbox exposed host to this LAN IP (obtained by freedombox)
4. send me a mail (or IRC Message) and tell me your Internet IP, I will check from the outside world which services are available and which are not

Now we should have a clean test setup for debugging and can start investigate what is wrong with Let's encrypt and DynDNS. 

Don't misunderstand, I don't say your current setup is not relevant nor I want to advise you to throw away your OpenWRT box. I just want to exclude as many potential sources of errors. After we found the main issue, you can set up everything as it was before. Is that OK for you?

**Comment by mray** [comment]
Your hints were very helpful!

I connected the FBX directly to my Fritzbox and could instantly start generating the Let's Encrypt certificate. That is why I found out that the exposed host IP wasn't valid from FritzBoxes view; it is running its own DHCP server. I still don't know how this is possible but both my routers run DHCP and assign IPs to each other.
I THEN found out that by typing (instead of copy+paste from my password manager) I mixed up `l` and `I` in the password. Total #FAIL on my part, sorry.
After correctly exposing all ports to my OpenWRT and a forward of TCP ports 80 and 443 to FreedomBox everything worked fine finally. Thank you very much for the patience.

As a final thought on the matter: I'm still concerned about two things:
 1. I assume many people (the majority) do run FBX behind a router
 2. DynDNS, Port forwarding and certificate generation is mandatory to use FBX with key features of Address and Calendar services.
We still need to simplify that about orders of magnitude to make it *just work*.

These things might be sensible steps:
* provide templates for known DynDNS server configurations to ease the use
* offer guide and help in finding alternative ways to make your box reachable from the outside (Tor IDs and free DynDNS)
* install Let's Encrypt & use certificates automatically
* help people understand what they need to do on their local router to forward the needed ports.
